### ISSUE TITLE
LDClient `identify` completion handler is never invoked when there are network connectivity issues

### ISSUE THREAD

**Issue by SD10** [launchdarkly/ios-client#264]
**Describe the bug**
The `identify` method of `LDClient` never invokes its `completion` block parameter in situations of poor connectivity that cause the underlying network requests to fail

**To reproduce**
- Add LaunchDarkly 5.4.5 to a new project
- Use the `LDClient.identify` method and provide a `completion` block
- Simulate 100% loss of network connection
- Run the project

```swift
let user = LDUser(key: "1")
LDClient.get().identify(user) {
    print("Call me maybe")
}
```

**Expected behavior**
The `identify` method should call its `completion` handler after a specified timeout and cause the `print` to be executed. 

**Logs**
- The console shows the underlying requests failing and the LDClient trying to reconnect infinitely
- The text from the print statement is never logged

**SDK version**
5.4.5

**Language version, developer tools**
Xcode 13, Swift 5

**OS/platform**
iOS 15

**Additional context**
The [LDClient documentation](https://launchdarkly.github.io/ios-client-sdk/Classes/LDClient.html) states:
>  A completion may be passed to the identify method to allow a client app to know when fresh flag values for the new user are ready.

and

> completion: Closure called when the embedded setOnlineIdentify call completes, subject to throttling delays. (Optional)

This makes it dangerous to block user progression in the application using the `identify` method since it will never timeout in the case of network connectivity issues or issues with the Launch Darkly service.

1. Adding a `timeout` parameter to the `identify` method to would help mitigate this issue
2. Adding additional context to the `completion` handler such as a boolean flag indicating whether the `completion` handler is being invoked in the scenario of success or failure would be very useful for error handling as well

Related issues: #211, #209 

Thank you 🙏

**Comment by louis-launchdarkly** [comment]
Hello @SD10, thank you for reaching out, and sorry for the late response - from your description and the related issues, we could be missing some connectivity handling when you call `identify`. Let me discuss this with the team and get back to you once we have a better understanding of the issue.

**Comment by taraspasichnyk** [comment]
Our team experiences that as well. Are there any updates?

**Comment by louis-launchdarkly** [comment]
Hello @SD10 and @taraspasichnyk, thank you for reaching out.

We are still investigating, but have not yet determined the best course of action.

Ideally, the API would both lend itself to correct usage and also allow differentiation between a successful identify and a timeout. Also, we may need additional handling that allows us to notice when a condition is likely not going to recover in a timely manner.

We will provide an update, either in guidance for an alternate way to use the API or as an SDK update, when we can.

**Comment by simondelphia** [comment]
There's now a `startWaitSeconds` parameter, and the `completion` block has a `timedOut` parameter as well. So presumably this can be closed?

**Comment by fred-riverside** [comment]
Same issue here at Riverside...

**Comment by YoavGro** [comment]
@louis-launchdarkly any update on this?

**Comment by keelerm84** [comment]
In the recently released v9.4.0, we introduced a new `identify` method which accepts a timeout interval. If an identify request does not complete within that time frame, the provided callback will be triggered with a result value of `.timeout`.
